br/pkg/lightning/backend/local/region_job.go

```diff
@@ -41,12 +42,39 @@ import (
 
 type jobStageTp string
 
-// nil -> regionScanned: create a new region job
-// regionScanned -> wrote: write the data to TiKV
-// wrote -> ingested: ingest the data to TiKV
-// ingested -> nil: finish the job
-// regionScanned / wrote -> needRescan: need to rescan the data, maybe region is expanded.
-// needRescan -> nil: discard the job. caller will create a new job from unfinishedRanges.
+/*
+	       +
+	       v
+	+------+------+
+
++->+regionScanned+<------+
+|  +------+------+       |
+|         |              |
+|         |              |
+|         v              |
+|      +--+--+     +-----+----+
+|      |wrote+---->+needRescan|
+|      +--+--+     +-----+----+
+|         |              ^
+|         |              |
+|         v              |
+|     +---+----+         |
++-----+ingested+---------+
+
+	+---+----+
+	    |
+	    v
+
+above diagram shows the state transition of a region job, here are some special
+cases:
+  - regionScanned can directly jump to ingested if the keyRange has no data
+  - regionScanned can only transit to wrote. TODO: check if it should be transited
```

Comment:
@gozssky in release-6.5, if lightning meet "region is changed" error, like region split, `WriteToTiKV` will return a !isRetryableImportTiKVError error, and it will break the whole import procedure? Will TiKV side return "region is changed" error during ImportSstService.write?